test/connection.js

```diff
@@ -622,7 +622,6 @@ describe('Connection', () => {
 
                             expect(err).to.not.exist();
                             expect(count1).to.equal(2);
-                            expect(Object.keys(server.connections[0]._connections).length).to.equal(2);
```

Comment:
You should actually create the socket and `TLS.connect()` one at a time to be safe (rather than creating/connecting both at the same time then calling `TLS.connect()` in series).